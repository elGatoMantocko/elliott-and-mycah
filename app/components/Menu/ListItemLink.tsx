import ListItem, { ListItemProps } from '@material-ui/core/ListItem';
import * as React from 'react';
import { useHistory } from 'react-router';

export const ListItemLink = ({
  disabled,
  href,
  onClick,
  children,
}: ListItemProps<'a', { button?: true }>) => {
  const { push } = useHistory();
  return (
    <ListItem
      button
      disabled={disabled}
      component="a"
      onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
        if (disabled === true) {
          return e.preventDefault();
        }
        href != null && push(href);
        onClick != null && onClick(e);
      }}
    >
      {children}
    </ListItem>
  );
};
