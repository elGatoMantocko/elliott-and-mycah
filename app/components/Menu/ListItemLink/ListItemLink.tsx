import { ListItemButton } from '@mui/material';
import React from 'react';
import { Link as RouterLink, LinkProps } from 'react-router-dom';

interface ListItemLinkProps extends LinkProps {
  disabled?: boolean;
}
export const ListItemLink = ({ disabled, children, ...props }: ListItemLinkProps) => (
  <ListItemButton {...props} component={RouterLink} disabled={disabled}>
    {children}
  </ListItemButton>
);
