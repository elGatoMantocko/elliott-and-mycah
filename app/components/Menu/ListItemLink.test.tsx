import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';

import { ListItemLink } from './ListItemLink';

describe('<ListItemLink />', () => {
  it('should render a link', () => {
    const el = render(<ListItemLink href="/test">Test</ListItemLink>, { wrapper: MemoryRouter });
    expect(el.container.innerHTML).toMatchSnapshot();
  });

  it('should render a disabled link', () => {
    const el = render(
      <ListItemLink disabled href="/test">
        Test
      </ListItemLink>,
      { wrapper: MemoryRouter },
    );
    expect(el.container.innerHTML).toMatchSnapshot();
  });
});
