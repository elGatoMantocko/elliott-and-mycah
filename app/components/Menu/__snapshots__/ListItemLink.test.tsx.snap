// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemLink /> should render a disabled link 1`] = `
<div>
  <a
    aria-disabled="true"
    class="MuiListItemButton-root MuiListItemButton-gutters Mui-disabled MuiButtonBase-root Mui-disabled css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="-1"
  >
    Test
  </a>
</div>
`;

exports[`<ListItemLink /> should render a link 1`] = `
<div>
  <a
    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="0"
  >
    Test
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</div>
`;
